python_modules/libraries/dagster-fivetran/dagster_fivetran/assets.py

```diff
@@ -0,0 +1,99 @@
+from typing import List, Optional
+
+from dagster import AssetKey, Out, Output
+from dagster.core.asset_defs import AssetsDefinition, multi_asset
+from dagster.utils.backcompat import experimental
+from dagster_fivetran.resources import DEFAULT_POLL_INTERVAL
+from dagster_fivetran.utils import generate_materializations
+
+
+@experimental
+def fivetran_assets_factory(
+    connector_id: str,
+    asset_keys: List[AssetKey],
+    name: Optional[str] = None,
+    poll_interval: float = DEFAULT_POLL_INTERVAL,
+    poll_timeout: Optional[float] = None,
+    io_manager_key: Optional[str] = None,
+) -> AssetsDefinition:
+
+    """
+    Create a @multi_asset for a given Fivetran connector.
+
+    Returns an AssetsDefintion which connects the specified ``asset_keys`` to the computation that
+    will update them. Internally, executes a Fivetran sync for a given ``connector_id``, and
+    polls until that sync completes, raising an error if it is unsuccessful. Requires the use of the
+    :py:class:`~dagster_fivetran.fivetran_resource`, which allows it to communicate with the
+    Fivetran API.
+
+    Args:
+        connector_id (str): The Fivetran Connector ID that this op will sync. You can retrieve this
+            value from the "Setup" tab of a given connector in the Fivetran UI.
+        asset_keys (str): The set of asset keys that Dagster will expect this asset to produce.
+            These should be of the form ``AssetKey([<schema_name>, <table_name>])`` where
+            ``<schema_name>`` and ``<table_name>`` represent the location of the tables in the
+            Fivetran destination. Any AssetKey listed here MUST be sync'd every time this computation
+            is executed, otherwise an exception will be raised. Additional tables that are updated
+            during the sync but not listed here will be recorded with runtime AssetMaterialization
+            events.
```

Comment:
Maybe another sentence about the consequence of this - that explicitly declared assets will always appear in the asset catalog but runtime ones might appear and disappear depending on what's happening upstream.